src/Makefile.am

```diff
@@ -707,6 +707,9 @@ uninstall-local: squid.conf.default
 	@$(SHELL) $(top_srcdir)/scripts/remove-cfg.sh "$(RM)" $(DESTDIR)$(DEFAULT_MIME_TABLE) $(srcdir)/mime.conf.default
 	@$(SHELL) $(top_srcdir)/scripts/remove-cfg.sh "$(RM)" $(DESTDIR)$(DEFAULT_CONFIG_FILE) squid.conf.default
 
+distclean-local:
+	test -d "cf_gen.dSYM" && rm -rf "cf_gen.dSYM" || $(TRUE)
```

Comment:
> The cf_gen target may leak this subdirectory for cf_gen

What makes cf_gen special? In other words, why do we need to add this special cleanup for cf_gen but not for other binaries? Perhaps the problem should be addressed by modifying how cf_gen is built?

Reply:
I think it could be be because it's the only noinst_PROGRAM we are buildingq

Reply:
Does temporary making cf_gen an installed program solves the problem (without this PR changes)?

Reply:
> Does temporary making cf_gen an installed program solves the problem (without this PR changes)?

Even if it did pass the build test, it would make no sense to keep it in, because by doing so it would cause cf_gen to be installed to a target system. The only reason to do it would be to possibly file a bug against autoconf

Reply:
Okay, I figured it out.
TL;DR: we override too much of autotools to build cf_gen. Updates coming up